com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -1265,6 +1281,63 @@ static void CheckAndApplyDebugSettings(ref RenderingData renderingData)
             }
         }
 
+        /// <summary>
+        /// Returns the best supported image upscaling filter based on the provided upscaling filter selection
+        /// </summary>
+        /// <param name="imageSize">Size of the final image</param>
+        /// <param name="renderScale">Scale being applied to the final image size</param>
+        /// <param name="selection">Upscaling filter selected by the user</param>
+        /// <returns>Either the original filter provided, or the best replacement available</returns>
+        static ImageUpscalingFilter ResolveUpscalingFilterSelection(Vector2 imageSize, float renderScale, UpscalingFilterSelection selection)
+        {
+            // By default we just use linear filtering since it's the most compatible choice
+            ImageUpscalingFilter filter = ImageUpscalingFilter.Linear;
+
+            switch (selection)
+            {
+                case UpscalingFilterSelection.Auto:
+                {
+                    // The user selected "auto" for their upscaling filter so we should attempt to choose the best filter
+                    // for the current situation. When the current resolution and render scale are compatible with integer
+                    // scaling we use the point sampling filter. Otherwise we just use the default filter (linear).
+                    float pixelScale = (1.0f / renderScale);
+                    bool isIntegerScale = ((pixelScale - Mathf.Floor(pixelScale)) == 0.0f);
+
+                    if (isIntegerScale)
+                    {
+                        float widthScale = (imageSize.x / pixelScale);
+                        float heightScale = (imageSize.y / pixelScale);
+
+                        bool isImageCompatible = (((widthScale - Mathf.Floor(widthScale)) == 0.0f) &&
+                                                  ((heightScale - Mathf.Floor(heightScale)) == 0.0f));
+
+                        if (isImageCompatible)
+                        {
+                            filter = ImageUpscalingFilter.Point;
+                        }
+                    }
+
+                    break;
+                }
+
+                case UpscalingFilterSelection.Linear:
+                {
+                    // Do nothing since linear is already the default
+
+                    break;
+                }
+
+                case UpscalingFilterSelection.Point:
+                {
+                    filter = ImageUpscalingFilter.Point;
+
+                    break;
+                }
```

Comment:
I don't immediately understand the benefit for expose this to users, in which cases they would select bilinear or point instead of Auto? Or is this a preparation for FSR scaling mode in the follow up PR?

Reply:
Linear and Point were both added before I actually had figured out how to dynamically switch between them based on the current screen size and render scale like in Auto. I figured we'd keep them in case users are unhappy with the automatic filter's behavior and want to explicitly control it themselves? I could also see how this might make things more confusing though.

Yes, FSR does slot into this enum in the next PR so that's the main reason for it.

Reply:
Ok, yeah if you think it might be useful it's ok, I don't fully understand what could be the cases they are unhappy thought. If the difference between the two modes end up being artistic choice then this makes sense, but if they are unhappy because auto is not working as expected that kind of implies to me an issue that should be considered before we make this decision.

Reply:
Yeah my thought was that some people may not artistically like the integer scaling behavior from auto. It's sharper but with more visible pixelation? Some people may just prefer the blur from linear instead. (The pixelation depends on display pixel density though)

We'll want to make sure the Auto filter's behavior is clearly documented for users so they're aware that it jumps to integer scaling when at appropriate scale + resolution combos.